x-pack/plugins/alerting/server/lib/last_run_status.ts

```diff
@@ -42,9 +51,22 @@ export const lastRunFromState = (stateWithMetrics: RuleTaskStateAndMetrics): ILa
     outcomeMsg = translations.taskRunner.warning.maxExecutableActions;
   }
 
+  // Overwrite outcome to be error if last run reported any errors
+  if (errors.length > 0) {
+    outcome = RuleLastRunOutcomeValues[2];
+  }
+
+  // Optionally merge outcome messages reported by
+  // rule execution to the Framework's outcome message
+  if (outcomeMsg.length > 0 && outcomeMessage.length > 0) {
+    outcomeMsg = `${outcomeMsg} - ${outcomeMessage}`;
+  } else if (outcomeMessage.length > 0) {
+    outcomeMsg = outcomeMessage;
+  }
+
   return {
     lastRun: {
-      outcome,
+      outcome: outcome || RuleLastRunOutcomeValues[0],
```

Comment:
Nit: `||` is probably unnecessary

Reply:
I agree since it has been initialized with this value